### PR TITLE
Add static code analysis submission to CI

### PR DESCRIPTION
**Pull request by zabbal**
Test run: https://main.gitlab.in.here.com/olp/edge/ota/connect/client/aktualizr/pipelines/49815
Note: it's expected to be used via "schedule" on a weekly basis, not a regular pipeline.

### CODE REVIEW

**Comment by lbonn** [comment]
The one you linked does not contain the veracode step, I think you meant https://main.gitlab.in.here.com/olp/edge/ota/connect/client/aktualizr/pipelines/48773

**Comment by zabbal** [comment]
> The one you linked does not contain the veracode step, I think you meant https://main.gitlab.in.here.com/olp/edge/ota/connect/client/aktualizr/pipelines/48773

Yes, definitely. The difference is in how it's triggered - if it's not via schedule than the step is ignored. Thanks, I'll update comment.

**Comment by zabbal** [comment]
Not sure about failed CI - when I follow the links the job is marked as successful. Any ideas how to re-trigger/fix this?

**Comment by lbonn** [comment]
This one doesn't look successful: https://main.gitlab.in.here.com/olp/edge/ota/connect/client/aktualizr/pipelines/50015, but it's because of an unrelated issue we have with gitlab.

Don't know why it was triggered after this one though: https://main.gitlab.in.here.com/olp/edge/ota/connect/client/aktualizr/pipelines/49815

File: ci/gitlab/.gitlab-ci.yml [hunk 1]
```diff
@@ -85,6 +85,49 @@ nop11:
...
+
+veracode-scan:
+  stage: deploy
+  only:
+    - schedules
+  dependencies:
+    - scan-build
+  image: "$UBUNTU_BIONIC_PR_IMAGE"
+  before_script:
+    - apt update
```
Comment (lbonn) [inline ci/gitlab/.gitlab-ci.yml]: It looks like you actually do not need our Docker image here, you could pull an image with the correct version of java preinstalled and not worry about installing it with apt in the pipeline.

**Comment by lbonn** [comment]
So, as I understand, this step is only triggering a job that's running somewhere else, right?

Do you think it would be possible to output a link to the triggered job somewhere in the logs, so that it can be inspected easily from here?

File: ci/gitlab/.gitlab-ci.yml (hunk 1 above)
Comment (zabbal) [reply ci/gitlab/.gitlab-ci.yml]: Which image is that? Got link/name?

**Comment by zabbal** [comment]
> So, as I understand, this step is only triggering a job that's running somewhere else, right?

That's correct.

> Do you think it would be possible to output a link to the triggered job somewhere in the logs, so that it can be inspected easily from here?

That won't help - you'll have to request account from SPC to view the scan results.

> If that's indeed the case, the job could also be renamed to highlight the fact that it's a trigger.

Any particular suggestions?

File: ci/gitlab/.gitlab-ci.yml (hunk 1 above)
Comment (lbonn) [reply ci/gitlab/.gitlab-ci.yml]: https://hub.docker.com/_/openjdk ?

**Comment by lbonn** [comment]
> > Do you think it would be possible to output a link to the triggered job somewhere in the logs, so that it can be inspected easily from here?
> 
> That won't help - you'll have to request account from SPC to view the scan results.

Well, that could help people with an SPC account, but that's not critical sure. Do they at least have some information about the commit used for the test there?

> 
> > If that's indeed the case, the job could also be renamed to highlight the fact that it's a trigger.
> 
> Any particular suggestions?

veracode-scan-trigger

File: ci/gitlab/.gitlab-ci.yml (hunk 1 above)
Comment (mike-sul) [reply ci/gitlab/.gitlab-ci.yml]: AFAIK, gitlab can serve as a docker image registry, we could prepare our own images, put them into gitlab so CI pipelines/test can utilize them. Effectively, each aktualizr docker file 'bakes' dynamically/on the fly during test running, while having pre-baked docker image could speed these tests little bit.

**Comment by lbonn** [comment]
Make another proposal on top of your changes here: https://github.com/advancedtelematic/aktualizr/pull/1281

**Comment by lbonn** [comment]
Replaced by #1281

**Pull request closed without merging**